ocs_ci/utility/metadata_utils.py

```diff
@@ -0,0 +1,258 @@
+"""
+Module that contains all operations related to add metadata feature in a cluster
+
+"""
+
+import logging
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources import pod
+
+log = logging.getLogger(__name__)
+
+
+def check_setmetadata_availability(pod_obj):
+    """
+    Check setmetadata parameter is available or not for cephfs and rbd plugin pods
+
+    """
+    plugin_provisioner_pod_objs = pod.get_all_pods(
+        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+        selector=["csi-cephfsplugin-provisioner", "csi-rbdplugin-provisioner"],
+    )
+    log.info(f"list of provisioner pods---- {plugin_provisioner_pod_objs}")
+    response = pod.validate_pods_are_respinned_and_running_state(
+        plugin_provisioner_pod_objs
+    )
+    log.info(response)
+    for plugin_provisioner_pod in plugin_provisioner_pod_objs:
+        args = pod_obj.exec_oc_cmd(
+            "get pod "
+            + plugin_provisioner_pod.name
+            + " --output jsonpath='{.spec.containers[4].args}'"
+        )
+        return "--setmetadata=true" in args
+
+
+def enable_metadata(
+    config_map_obj,
+    pod_obj,
+):
+    """
+    Enable CSI_ENABLE_METADATA
+    Steps:
+    1:- Enable CSI_ENABLE_METADATA flag via patch request
+    2:- Check csi-cephfsplugin provisioner and csi-rbdplugin-provisioner
+    pods are up and running
+    3:- Check 'setmatadata' is set for csi-cephfsplugin-provisioner
+    and csi-rbdplugin-provisioner pods
+
+    Returns:
+        str: cluster name
+
+    """
+    enable_metadata = '{"data":{"CSI_ENABLE_METADATA": "true"}}'
+
+    # Enable metadata feature for rook-ceph-operator-config using patch command
+    assert config_map_obj.patch(
+        resource_name="rook-ceph-operator-config",
+        params=enable_metadata,
+    ), "configmap/rook-ceph-operator-config not patched"
+
+    # Check csi-cephfsplugin provisioner and csi-rbdplugin-provisioner pods are up and running
+    assert pod_obj.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector="app=csi-cephfsplugin-provisioner",
+        dont_allow_other_resources=True,
+        timeout=60,
+    )
+
+    assert pod_obj.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector="app=csi-rbdplugin-provisioner",
+        dont_allow_other_resources=True,
+        timeout=60,
+    )
+
+    # Check 'setmatadata' is set for csi-cephfsplugin-provisioner and csi-rbdplugin-provisioner pods
+    res = check_setmetadata_availability(pod_obj)
+    if not res:
+        raise AssertionError
+
+    cephfsplugin_provisioner_pods = pod.get_cephfsplugin_provisioner_pods()
+
+    args = pod_obj.exec_oc_cmd(
+        "get pod "
+        + cephfsplugin_provisioner_pods[0].name
+        + " --output jsonpath='{.spec.containers[4].args}'"
+    )
+    for arg in args:
+        if "--clustername" in arg:
+            return arg[14:]
```

Comment:
please add a comment explaining what this code does

Reply:
Addressed in latest commit.